### PR TITLE
Fix custom loss for MXNet backend. Fix bug in Concat layer

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
* Fix bug in Concat operator leading to failure in Layers.Concatenate.
* Fix shape mis-match errors when using Custom Loss.
* Enable 6 more unit tests.

@roywei @kalyanee

### CODE REVIEW

File: keras/engine/training.py [hunk 1]
```diff
@@ -437,13 +437,20 @@ def weighted(y_true, y_pred, weights, mask=None):
...
+
         if weights is not None:
             # reduce score_array to same ndim as weight array
             ndim = K.ndim(score_array)
             weight_ndim = K.ndim(weights)
             score_array = K.mean(score_array, axis=list(range(weight_ndim, ndim)))
+            # If sample_weights shape is like (100, ), we convert it to (100, 1).
+            # Because, MXNet treats the shape (100, ) as (100) leading to broadcast operator
+            # failures in below operations.
+            if K.backend() == 'mxnet' and K.ndim(weights) == 1:
```
Comment (roywei) [inline keras/engine/training.py]: we can use `weight_ndim` already calculated:
```
if K.backend() == 'mxnet' and weight_ndim== 1:
```

File: tests/keras/layers/normalization_test.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def test_batchnorm_correctness_2d():
 
 
 @pytest.mark.skipif((K.backend() == 'mxnet'),
-                    reason='MXNet backend does not allow predict() before compile()')
+                    reason='MXNet backend uses native BatchNorm operator. Donot do updates in the model.')
```
Comment (roywei) [inline tests/keras/layers/normalization_test.py]: Do not?

File: tests/keras/layers/normalization_test.py [hunk 3]
```diff
@@ -160,7 +160,7 @@ def test_batchnorm_convnet_no_center_no_scale():
 
 
 @pytest.mark.skipif((K.backend() == 'mxnet'),
-                    reason='MXNet backend uses native BatchNorm operator. Do do updates in the model.')
+                    reason='MXNet backend uses native BatchNorm operator. Donot do updates in the model.')
```
Comment (roywei) [inline tests/keras/layers/normalization_test.py]: same - Do not?

File: keras/engine/training.py (hunk 1 above)
Comment (sandeep-krishnamurthy) [reply keras/engine/training.py]: Done. Thanks

File: tests/keras/layers/normalization_test.py (hunk 2 above)
Comment (sandeep-krishnamurthy) [reply tests/keras/layers/normalization_test.py]: Done

File: tests/keras/layers/normalization_test.py (hunk 3 above)
Comment (sandeep-krishnamurthy) [reply tests/keras/layers/normalization_test.py]: Done

**Comment by sandeep-krishnamurthy** [comment]
@roywei - Fixed your comments

**Pull request merged**